clusterloader2/pkg/measurement/common/api_availability_metrics.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"time"
+)
+
+type apiAvailabilityMetrics struct {
+	numSuccesses               int
+	numFailures                int
+	maxConsecutiveFailedProbes int
+	consecutiveFailedProbes    int
+}
+
+type apiAvailabilityOutput struct {
```

Comment:
Actually, I would prefer the output to be:

```
type apiAvailabilitySummary struct {
  IP                                           string
  AvailabilityPercentage           float32
  LongestUnavailableDuration time.Duration
}

type apiAvailabilityOutput struct {
  ClusterMetrics apiAvailabilitySummary
  MasterMetrics []apiAvailabilitySummary
}
```

Reply:
hi @wojtek-t , slight confusion here. The final output for MasterMetrics is an array here. If you see the output of the measurement that would be returned, all outputs would be consolidated into 1. 

https://github.com/kubernetes/perf-tests/pull/1162/files#diff-769e85d1bdfa456cbbe72288f2ba2ab5R54-R55

Do you think this isn't similar to what you prefer?

Reply:
I care about 'availability' summary as a whole. Yes - I want a single summary, instead of N separate ones.
Alternatively, we may consider two measurements: one for the cluster, second for all masters.

Reply:
I see. I might want to make some tweaks to the apiAvailabilityMeasurement struct in that case. Currently it looks like this:

```
type apiAvailabilityMeasurement struct {
	isRunning           bool
	stopCh              chan struct{}
	hosts               []string
	summaries           []measurement.Summary
	clusterLevelMetrics *apiAvailabilityMetrics
	hostLevelMetrics    map[string]*apiAvailabilityMetrics
	wg                  sync.WaitGroup
}
```
So, if I understand correctly, instead of an array of summaries, you want 1 cluster level summary and array of MasterLevel summaries in a single object. I would also need to remove the clusterLevelMetrics and hostLevelMetrics from the above struct.

Regarding 2 separate measurements, do we expose them as separate measurements- which seems slightly confusing to me. With a unified measurement, a user would be able to get all availability metrics in one summary.

Reply:
Sorry - I meant "two summaries" - one for the cluster, second for all masters.
But, I think having a single summary is cleaner.

Reply:
ok. so in that case, i will make the code changes to have one summary in the format you explained above.

Reply:
@wojtek-t I have made the code changes to expose only 1 summary for both ClusterLevel and MasterMetrics. Plz review when you get the chance..